source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -53,47 +53,52 @@ void AMDSClientTCPSocket::onSocketError(QAbstractSocket::SocketError error)
 
 void AMDSClientTCPSocket::readClientRequestMessage()
 {
+	QDataStream *inDataStream = 0;
+
 	if (tcpSocket_->bytesAvailable() < (int)sizeof(quint32))
 		return;
 
-	QDataStream *inDataStream = 0;
+	bool onResponseBoundary = false; // there might be multiple request in the buffer, we will read them out one by one
 	if (!waitingMorePackages_) {
+		readedBufferSize_ = 0;
+
 		inDataStream = new QDataStream(tcpSocket_);
```

Comment:
Memory leak here. If you read the entire request in one call, then you new the inDataStream here and also on line 98. You'll clean up the line 98 one, but nothing ever seems to clean this one up.